spec/models/ext_management_system_spec.rb

```diff
@@ -312,6 +312,40 @@
     end
   end
 
+  context "refresh" do
+    before do
+      @ems = FactoryBot.create(:ext_management_system)
+    end
+
+    it "raises an error if the authentication check fails" do
+      allow(@ems).to receive(:missing_credentials?).and_return(false)
+      allow(@ems).to receive(:authentication_status_ok?).and_return(false)
+
+      expect { @ems.refresh }.to raise_error(RuntimeError, "Provider failed last authentication check")
+    end
+
+    it "raises an error if no provider credentials are defined" do
+      allow(@ems).to receive(:authentication_status_ok?).and_return(true)
+      allow(@ems).to receive(:missing_credentials?).and_return(true)
+
+      expect { @ems.refresh }.to raise_error(RuntimeError, "no Provider credentials defined")
+    end
+
+    it "calls the EmsRefresh.refresh method internally" do
+      allow(@ems).to receive(:missing_credentials?).and_return(false)
+      allow(@ems).to receive(:authentication_status_ok?).and_return(true)
+      allow(EmsRefresh).to receive(:refresh)
+
+      @ems.refresh
+
+      expect(EmsRefresh).to have_received(:refresh)
+    end
+
+    it "defines a queue_refresh alias for refresh_ems" do
+      expect(@ems.method(:refresh_ems)).to eql(@ems.method(:queue_refresh))
```

Comment:
This test seems unnecessary to me as it just tests the implementation itself, and not the meaning behind it.  IMO, this should be a test that calls queue_refresh and ensure that a queue item appears with the correct contents.

Reply:
I'm confused. There's no implementation test here. This just checks that an alias exists, nothing more. I didn't add queue_refresh implementation tests because, well, I didn't create that method. :)

Reply:
I do realize it's somewhat out of place since it's not a `refresh` test, but I thought I'd toss it in there since there were no specs for `refresh_ems` specifically.